libraries/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -843,3 +843,38 @@ void IotMqtt_ReceiveCallback( IotNetworkConnection_t pNetworkConnection,
 }
 
 /*-----------------------------------------------------------*/
+
+IotMqttError_t IotMqtt_GetIncomingMQTTPacketTypeAndLength( IotMqttPacketInfo_t * pIncomingPacket,
+                                                           IotMqttGetNextByte_t getNextByte,
+                                                           void * pNetworkConnection )
+{
+    IotMqttError_t status = IOT_MQTT_SUCCESS;
+
+    /* Read the packet type, which is the first byte available. */
+    if( getNextByte( pNetworkConnection, &( pIncomingPacket->type ) ) == IOT_MQTT_SUCCESS )
+    {
+        /* Check that the incoming packet type is valid. */
+        if( _incomingPacketValid( pIncomingPacket->type ) == false )
+        {
+            IotLogError( "(MQTT connection) Unknown packet type %02x received.",
+                         pIncomingPacket->type );
+
+            status = IOT_MQTT_BAD_RESPONSE;
+        }
+        else
+        {
+            /* Read the remaining length. */
+            pIncomingPacket->remainingLength = _IotMqtt_GetRemainingLength_Generic( pNetworkConnection,
+                                                                                    getNextByte );
+
+            if( pIncomingPacket->remainingLength == MQTT_REMAINING_LENGTH_INVALID )
+            {
+                status = IOT_MQTT_BAD_RESPONSE;
+            }
+        }
+    }
+
+    return status;
```

Comment:
If `getNextByte(...) != IOT_MQTT_SUCCESS` on line 854, this function will return `IOT_MQTT_SUCCESS` yet `pIncomingPacket->remainingLength` will not have been set.

Reply:
Good catch. Added else clause to set status.